smithy-utils/src/main/java/software/amazon/smithy/utils/TypedCodeWriter.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.utils;
+
+/**
+ * Helper class for generating code.
+ *
+ * <p>This class is a general purpose implementation of {@link AbstractCodeWriter}.
+ */
+public class TypedCodeWriter extends AbstractCodeWriter<TypedCodeWriter> {
```

Comment:
nit: Can we call this `SimpleCodeWriter` or something similar? We alread have `SimpleParser`, so it's consistent.